### DISCUSSION TITLE
A document store loaded from file, filtered to a subset of files

### DISCUSSION THREAD

**Discussion by pelikhan** [run-llama/LlamaIndexTS#546] [Q&A]
Hello all, I was looking for the best way to instantiate a document store loaded from disk, for a set of files.

**Comment by pelikhan** [comment]
To be more precise, assuming I already have the documents indexed, I would llike to do a query on a subset of them without doing the full ingestion process.

**Comment by pelikhan** [comment]
I solved this by clone the vector store (fromDict, toDict), then selectively calling deleting on the unwanted docs.

```ts
            // get all documents
            const docs = await storageContext.docStore.getAllRefDocInfo()
            // reload vector store, no fs connection
            const vectorStore = await SimpleVectorStore.fromDict(
                await (
                    await SimpleVectorStore.fromPersistDir(persistDir)
                ).toDict()
            )
            // remove uneeded documents
            const toRemove = Object.keys(docs).filter(
                (id) => !files.includes(id)
            )
            for (const doc of toRemove) vectorStore.delete(doc)
            // swap in storateContext
            storageContext.vectorStore = vectorStore
 ``

**Comment by marcusschiesser** [comment]
@pelikhan, what does your query look like? Seems like it would make sense to add a parameter to the query to filter for the subset of docs.
How about adding metadata, see https://github.com/run-llama/LlamaIndexTS/blob/main/examples/chromadb/preFilters.ts ?
